packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -1534,6 +1534,22 @@ export function attach(
     return Array.from(knownEnvironmentNames);
   }
 
+  function isFiberHydrated(fiber: Fiber): boolean {
+    if (OffscreenComponent === -1) {
+      throw new Error('not implemented for legacy suspense');
+    }
+    switch (fiber.tag) {
+      case HostRoot:
+        const rootState = fiber.memoizedState;
+        return !rootState.isDehydrated;
+      case SuspenseComponent:
+        const suspenseState = fiber.memoizedState;
+        return suspenseState === null || suspenseState.dehydrated === null;
```

Comment:
I think maybe you should do `== null` to cover the case for code before we added the dehydrated property.

Reply:
I thought checking for `OffscreenComponent` earlier ensures this is only checked for modern Suspense? Or is the `dehydrated` in https://github.com/facebook/react/pull/16346 not always set for modern Suspense?